### ISSUE TITLE
Invalid log "logging event with no metadata" when there is metadata

### ISSUE THREAD

**Issue by petedoyle** [intercom/intercom-android#289]
## Version info
  - Intercom for Android version: 3.0.12
  - Android OS version: 7.0
  - Android Device: Nexus 6P

## Expected behavior
`logEvent(String name, Map<String, ?> metadata)` should log an event with the given metadata. No logcat error regarding empty metadata should be logged.

## Actual behavior
- A Logcat error occurs:
    - `The metadata provided is empty, logging event with no metadata`
- Debugging (on the decompiled .class in Android Studio) seems to indicate an errant `!`.
- Debugging (on the decompiled .class in Android Studio) seems to indicate the metadata is sent (e.g. the log is incorrect)
- The event and its metadata are correctly reported. (as observed in the Intercom admin interface)

## Steps to reproduce
 1. Call `Intercom.client().logEvent(String name, Map<String, ?> metadata)` with a Map<String, Object> containing at least one item.
 2. Observe logcat for errant message
 3. Confirm event sent with metadata

## Logs
```
AnalyticsManager  D  track(). event: AnalyticsEvent{name=My Event Name, properties={myId=101265}}
Intercom  I  The metadata provided is empty, logging event with no metadata
```

## Debugging Session
![image](https://cloud.githubusercontent.com/assets/68218/20387008/e92e97b2-ac72-11e6-979c-0563dabd4f50.png)

**Comment by ronocod** [comment]
Thanks for digging into this! We'll have a fix for this in the next version.

**Comment by ronocod** [comment]
Fixed in 3.0.13.
